docs/designs/reference_link.md

```diff
@@ -0,0 +1,194 @@
+> topics. So it's not recommended to include them in `docfx.yml`, they should be excluded from the initial scope `docfx.yml` if needed.
+
+### Include a [nested toc](table-of-contents.md#link-to-another-toc-file) using relative path
+
+The [toc](table-of-contents.md) syntax support to reference a nested toc using relative path or relative folder.
+
+For example, if `toc.md` reference `subfolder\toc.md`:
+
+```markdown
+#[child](subfolder\toc.md)
+```
+
+or 
+
+```markdown
+#[child](subfolder\)
+```
+
+All links in `subfolder\toc.md` are relative to the `subfolder\toc.md` itself, even when it's included by `toc.md`.
+
+### The other ways to link to a local resource
+
+#### Relative path starts with `~`
+
+Sometimes you may find it's complicated to calculate relative path between two files.
+DocFX also supports path starts with `~` to represent path relative to the root directory of your project (i.e. where `docfx.yml` is located).
+This kind of path will also be validated and resolved during build.
+
+For example, you can write the following links in `subfolder\file2.md` to reference `file1.md`:
+ 
+```markdown
+[file1](~/file1.md)
+
+[file1](../file1.md)
+```
+
+Both will be resolved to `../file1.html`.
+
+#### Absolute path starts with `\` or `/`
+
+It's recommended to always use relative path to reference another file in the same project. Relative path will be resolved during build and produce build warning if the target file does not exist.
+
+You can also use absolute path (path starts with `/`) to link to another file, but DocFX won't check its correctness for you and will keep it **as-is** in the output HTML.
+That means you should use the output file path as absolute path. For example, in the above case, you can also write the link as follows:
+
+```markdown
+[file2](/subfolder/file2.html)
+```
+
+#### Relative path in [inline HTML](https://daringfireball.net/projects/markdown/syntax#html)
+
+DocFX also supports to use relative path in inline HTML. Path in HTML link (`<a>`), image (`<img>`), script (`<script>`) and css (`<link>`) will also be resolved if they're relative path
+
+> [!Warning]
+> [Automatic link](https://daringfireball.net/projects/markdown/syntax#autolink) doesn't support relative path.
+> If you write something like `<file.md>`, it will be treated as an HTML tag rather than a link.
+
+## Link to a resource stored in [dependent repository](config.md)
+
+Besides using file path to link to a local resource, DocFX also supports to link a resource stored in [dependent repository](config.md)
+
+For example you have a dependent repository defined in config:
+
+```config
+dependencies:
+  dependent-name: https://github.com/dotnet/docfx-dependent#master
+```
+
+The folder structure in dependent repo is like below:
+
+```
+/
+|- subfolder/
+|  \- file2.md
+\- file1.md
+```
+
+You can link a resource stored in dependent repo:
+
+```markdown
+[dependency file1](dependent-name/file1.md)
```

Comment:
>dependent-name/file1.md [](start = 19, length = 23)

This path is dependent on the current file's path? or change it to `~/dependent-name/file1.md` to make it always right?

Reply:
~ represents the current docset, but the dependent file doesn't belong to current docset.


---
In reply to: [187233500](https://github.com/dotnet/docfx/pull/2752#discussion_r187233500) [](ancestors = 187233500)

Reply:
There is some existing user case that `~/` in dependent file represents the current docset's root. Can discuss this later.

---
In reply to: [187238214](https://github.com/dotnet/docfx/pull/2752#discussion_r187238214) [](ancestors = 187238214,187233500)